fvm/executionParameters.go

```diff
@@ -0,0 +1,137 @@
+package fvm
+
+import (
+	"github.com/onflow/cadence"
+	"github.com/onflow/cadence/runtime"
+	"github.com/onflow/cadence/runtime/common"
+
+	"github.com/onflow/flow-go/fvm/blueprints"
+	"github.com/onflow/flow-go/fvm/errors"
+	"github.com/onflow/flow-go/fvm/meter/weighted"
+	"github.com/onflow/flow-go/fvm/utils"
+)
+
+// getExecutionWeights reads stored execution effort weights from the service account
+func getExecutionEffortWeights(
+	env Environment,
+	service runtime.Address,
+) (
+	computationWeights weighted.ExecutionEffortWeights,
+	err error,
+) {
+	value, err := env.VM().Runtime.ReadStored(
+		service,
+		cadence.Path{
+			Domain:     blueprints.TransactionExecutionParametersPathDomain,
+			Identifier: blueprints.TransactionFeesExecutionEffortWeightsPathIdentifier,
+		},
+		runtime.Context{Interface: env},
+	)
+	if err != nil {
+		// this might be fatal, return as is
+		return nil, err
+	}
+
+	computationWeightsRaw, ok := utils.CadenceValueToWeights(value)
+	if !ok {
+		// this is a non-fatal error. It is expected if the weights are not set up on the network yet.
+		return nil, errors.NewCouldNotGetExecutionParameterFromStateError(
```

Comment:
If this is non-fatal - why even return error?

Reply:
In this context fatal errors are those that should crash the node, because something is very wrong.

In this case this is expected if the value is not set on the network yet. So a `errors.Error` is returned instead of a `error.Fatal`.

This error is currently handled by using default settings instead of those in the state.

In the future all networks should have these values defined in the state and this can be changed to a fatal error and we can remove the fallback to defaults.

Reply:
> In the future all networks should have these values defined in the state and this can be changed to a fatal error and we can remove the fallback to defaults.

Yes, thats my question - why use error instead of just default/empty values.

Reply:
> In the future all networks should have these values defined in the state and this can be changed to a fatal error and we can remove the fallback to defaults.

Yes, thats my question - why use error instead of just default/empty values.